components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -326,28 +302,40 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
         return;
     }
 
+    private String getStateParameter(AuthenticationContext context, Map<String, String> authenticatorProperties) {
+        String state = context.getContextIdentifier() + "," + OIDCAuthenticatorConstants.LOGIN_TYPE;
+        return getState(state, authenticatorProperties);
+    }
+
+    private String getOIDCCallbackUrl(Map<String, String> authenticatorProperties) {
+        String callbackurl = getCallbackUrl(authenticatorProperties);
+        if (StringUtils.isBlank(callbackurl)) {
+            callbackurl = IdentityUtil.getServerURL(FrameworkConstants.COMMONAUTH, true, true);
+        }
+        return callbackurl;
+    }
+
+    private String getOIDCAuthzEndpoint(Map<String, String> authenticatorProperties) {
+        String authorizationEP = getAuthorizationServerEndpoint(authenticatorProperties);
+
+        if (authorizationEP == null) {
```

Comment:
Shall we use StringUtils.isNotBlank() here?

Reply:
You mean StringUtils.isBlank()? Fxed.